File: temp/screenshot-slides.js
const { chromium } = require('playwright');

async function takeScreenshot(filePath, outputPath) {
  const browser = await chromium.launch({
    headless: true,
    args: ['--no-sandbox', '--disable-setuid-sandbox']
  });
  const page = await browser.newPage();
  await page.setViewportSize({ width: 960, height: 540 });

  await page.goto(`file://${filePath}`, { waitUntil: 'networkidle' });
  await page.screenshot({ path: outputPath, width: 960, height: 540 });

  await browser.close();
  console.log(`Screenshot saved: ${outputPath}`);
}

const slides = [
  { file: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-36.html', output: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-36.png' },
  { file: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-37.html', output: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-37.png' },
  { file: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-38.html', output: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-38.png' },
  { file: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-39.html', output: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-39.png' },
  { file: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-40.html', output: 'D:/新课开发/自然科学/23.地球科学/PPT/slides/slide-40.png' },
];

(async () => {
  for (const slide of slides) {
    await takeScreenshot(slide.file, slide.output);
  }
  console.log('All screenshots taken!');
})();

File: temp/screenshot_slides.py
from playwright.sync_api import sync_playwright
import os

output_dir = 'D:/新课开发/经营/系列/10_管理易筋经——计划、组织、领导、控制/slides'
slides = ['slide-83.html', 'slide-84.html']

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page(viewport={'width': 1280, 'height': 720})

    for slide in slides:
        file_path = os.path.join(output_dir, slide)
        page.goto(f'file:///{file_path}')
        page.wait_for_timeout(500)
        screenshot_path = os.path.join(output_dir, slide.replace('.html', '.png'))
        page.screenshot(path=screenshot_path, full_page=False)
        print(f'Screenshot saved: {screenshot_path}')

    browser.close()
